### ISSUE TITLE
Panic when zooming out a plot too far

### ISSUE THREAD

**Issue by c-git** [emilk/egui#3462]
I wasn't able to check if it is present on main by reproducing it but the [code that causes](https://github.com/emilk/egui/blob/fd75adb3a0e5f8ca373ac7e5e03f750e690ae87b/crates/egui_plot/src/lib.rs#L1877) the panic is still there.

**Describe the bug**

I was experimenting with zooming using the right mouse button. And while doing so seems I hit an edge case and the program suddenly closed. Checking the console I discovered that it had panicked. Seem I managed to accidentally try to zoom by exactly 0.0 and there is an assert to check that the amount is not 0.0 hence it crashed. While this is hard to recreate it is not impossible, might be worth adding a check before calling [`next_power`](https://github.com/emilk/egui/blob/master/crates/egui_plot/src/lib.rs#L1877) so that it doesn't panic but instead does nothing as that seems to make sense for a zoom by 0.

Error message:

```
thread 'main' panicked at /home/one/.cargo/registry/src/index.crates.io-6f17d22bba15001f/egui_plot-0.23.0/src/lib.rs:1877:5:
assertion `left != right` failed
  left: -0.0
 right: 0.0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**To Reproduce**
Given the nature of the bug I was unable to reproduce it as it requires zooming on a plot with the right mouse button and getting exactly 0 for the amount of zoom.

**Expected behavior**

I think nothing should happen instead of a panic as the user may trigger this (even though it is unlikely)

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.01
 - Browser: N/A running native when crashed
 - Version: 23

**Comment by Its-Just-Nans** [comment]
I think this could be closes with #4637 ?!

**Comment by emilk** [comment]
> I think this could be closes with #4637 ?!

No - I can still reproduce on egui.rs after zooming out for a minute or so
